taxonomy-editor-frontend/src/pages/editentry/ListTranslations.jsx

```diff
@@ -18,7 +19,9 @@ const ListTranslations = ({ nodeObject, setNodeObject }) => {
     const [mainLangRenderedTranslations, setMainLangRenderedTranslations] = useState([]) // Stores state of main language's tags
     const [openDialog, setOpen] = useState(false); // Used for Dialog component
     const [newLanguageCode, setNewLanguageCode] = useState(''); // Used for storing new LC from Dialog
-    const [isValidLanguageCode, setisValidLanguageCode] = useState(false); // Used for validating a new LC
+    const [languageAction, setLanguageAction] = useState('invalid'); // Used for storing state of action to be performed on LC (add/show/invalid)
```

Comment:
wondering if we could remove the `languageAction`. Or if too complex, at least do not allow the `invalid` state to happen, if that's too complex, guess it's ok as is.  
Let's say, for example, if we replace  the autocomplete with  a <select /> (or it's Material-UI equivalent), then only valid options can be selected by the user. 
Also, not understanding why 'invalid' is the default state, like why is it invalid even before doing anything.
Besides that, wondering if we could just have 'show' instead of 'show' and 'add'. I may be wrong, but my guess is that  from the user perspective there's not much difference, my understanding here is that it would make the user to first 'add' and then 'show' for a particular language? I mean, if the difference between both is not relevant to the user, can we hide it?

Reply:
1. The state is invalid in the beginning to hide the submission button (hide/show) until a valid selection is made.
2. **I think a good reason for keeping the button texts as show and add could be to tell the user that they are adding a node to a taxonomy (everyone will see it) vs when they are just showing a node (only added to their local preferences).**
This is why I did a three state toggle in the first place.
3. Otherwise, the state can toggle between show and invalid (true/false)

Reply:
May be missing something , but from a user perspective it seems that we are only adding allowing the user to show more languages, and seems to me that the difference between add and show is just if the language already has information, if that's the case I think we can hide that diference to the user. If that's not the case I'd worry we are mixing different features in the same dialog.